lnwire/update_add_htlc.go

```diff
@@ -78,6 +78,19 @@ type UpdateAddHTLC struct {
 	ExtraData ExtraOpaqueData
 }
 
+// BlindingPointOrNil returns the blinding point associated with the update, or
+// nil.
+func (c *UpdateAddHTLC) BlindingPointOrNil() *btcec.PublicKey {
```

Comment:
So this can just be `c.BlindingPoint.UnwrapOr(nil)`.

Reply:
`cannot use nil as tlv.RecordT[*tlv.tlvType0, *secp256k1.PublicKey] value in argument to htlc.BlindingPoint.UnwrapOr`? 

iiuc we need two layers of unwrapping here to get the underlying type in the record rather than the type in the option?

Reply:
Ah sure, had just glanced at that. Can also use `WhenSomeV` instead here to avoid needing to deref `Val`, but non-blocking ofc. 

Philosophical point, but generally I like to review the option as sort of lazy evaluation, so you pipe it down all the way until you actually need to obtain the true value. Otherwise, by forcing evaluation sooner than needed, you just end up burdening the caller once again with required nil checks.